velox/functions/prestosql/window/Rank.cpp

```diff
@@ -70,29 +70,31 @@ class RankFunction : public exec::WindowFunction {
       auto start = rawPeerStarts[i];
       if (start != currentPeerGroupStart_) {
         currentPeerGroupStart_ = start;
-        rank_ += previousPeerCount_;
+        if constexpr (type == RankType::kDenseRank) {
+          rank_ += (previousPeerCount_ != 0);
```

Comment:
Can previousPeerCount_ be 0 anytime here ? Every partition will have atleast one row. The first row doesn't satisfy this condition and so atleast one row has a rank set before. Which implies previousPeerCount_ is always atleast one. So always incrementing here by 1 is sufficient I feel. wdyt ?

Reply:
Since previousPeerCount_ is initialized to 0, it will be 0 for the first peer group so denseRank will be updated to 2 if we increment by 1 each time. The tests fail as a result so this check is necessary.